### DISCUSSION TITLE
Looking for feedback for version 2.0 🙌

### DISCUSSION THREAD

**Discussion by fgatti675** [firecmsco/firecms#193] [Ideas]
Hi everyone!

We are currently in the final stages of the beta period of version 1.0.0 of FireCMS and are almost ready for a RC.

While we iron out the last details for this release, we are already thinking of how the next version is going to look like.
The main feature of version 2.0 will be an **in-app schema and collections editor**, so you will be able to define your configuration without specifying it in code.

This is great, but it means we need to be able to serialise schemas and collections, so we can't persist functions such as permission builders or property builders.

We will we decoupling collections from schemas and now you will specify a `schemaId` in the collection (instead of passing a `schema`, like until now). Schemas will have an id and will be set as independent props.

We will take the chance to improve on the current API. So what I would like to ask you is: 

**What are the things you like less about setting up a FireCMS project right now?** 

For example, we are thinking about removing the `config` prop or properties and move all its fields to the parent properties, since now some parameters are found in the config and some in the property, which does not make a lot of sense.

Looking forward to your feedback!!

**Comment by mauriceackel** [comment]
Hi @fgatti675 I think I would like to see 2 features:

1. Being able to set a custom ID but **at the same time** being able to use an automatically created ID
2. For me the way to specify default values right now feels a bit odd, I would love to see the default  value property added to the individual property definitions instead of having it on the top level of the schema

**Reply by fgatti675** [reply]
Taking note, thanks!

**Comment by Jerunnon** [comment]
Hi @fgatti675 ,

It would be great to be able to create a date and time picker as single fields. Currently it seems like it is the DateTimePicker from MUI. They also offer Date and Time picker seperatly. One benefit would be to configure the Localization of the time picker.

**Reply by Jerunnon** [reply]
Oh and I also I saw that there is no `<MobileDateTimePicker> `or `<DesktopDateTimePicker>`. I haven't tested it on mobile yet, but would be a great addition, I think.

**Reply by fgatti675** [reply]
Adding this as a feature request for version 2 :)

**Comment by dhazeley** [comment]
I have 2 main requests that I think would make FireCMS much more useable and extensible for real world use cases.

1. Better support for collection groups and subcollection references. There's no easy way to create a view that pulls from a collection group without custom code. Even in a regular collection view, you cannot have a reference field that directs to a subcollection without custom code. These are 2 scenarios that are highly common (and suggested data model patterns by Google) that I think should be solvable through configurations.

2. Better modularity in the library's field components. When creating custom fields or views, it would be nice to use the FireCMS field components to maintain design uniformity and ease of development. The need for a (admittedly simple) wrapper and lack of documentation for the [Reference Field component](https://github.com/Camberi/firecms/blob/master/src/form/fields/ReferenceField.tsx) tells me that these components were not intended to be used outside the core library. If all the [form field components](https://github.com/Camberi/firecms/tree/master/src/form/fields) could be updated and documented for use outside the core library, that would be AMAZING.

**Comment by Jerunnon** [comment]
Also it would be nice if you could set your own colors, as an example, for catagories. Maybe I missed the functionalitym but would be a very nice feature.

**Reply by fgatti675** [reply]
Yes, for categories you can change the colors! Instead of asigning a string as a value, you can pass an object with the desired color

**Comment by warrenrhodes** [comment]
Hello I am a beginner in fireCms and I would like to know if there are executable functions when launching the cms. I would like to get a list of data in a collection and display it in another sub-collection.
